va/caa.go

```diff
@@ -40,12 +40,23 @@ func (va *ValidationAuthorityImpl) IsCAAValid(
 func (va *ValidationAuthorityImpl) checkCAA(
 	ctx context.Context,
 	identifier core.AcmeIdentifier) *probs.ProblemDetails {
-	present, valid, err := va.checkCAARecords(ctx, identifier)
+	present, valid, records, err := va.checkCAARecords(ctx, identifier)
 	if err != nil {
 		return probs.DNS("%v", err)
 	}
-	va.log.AuditInfof("Checked CAA records for %s, [Present: %t, Valid for issuance: %t]",
-		identifier.Value, present, valid)
+
+	// Build a string with the DNS presentation format of each *dns.CAA record we
+	// checked (if any) separated by commas to be included in the audit log line
+	var recordsStr strings.Builder
+	for i, rec := range records {
+		fmt.Fprintf(&recordsStr, "%q", rec.String())
```

Comment:
This is a bit complicated, could we just use a `[]string` and join them later with a `strings.Join`?

Reply:
@jsha's suggestion to use JSON resolved this feedback as well. See 7556d5d